### PR TITLE
Improves "Introduction to Chainer"

### PR DESCRIPTION
**Pull request by mitmul**
Update the very initial tutorial in the official documentation.

### CODE REVIEW

File: docs/source/tutorial/basic.rst [hunk 1]
```diff
@@ -112,9 +119,19 @@ In order to preserve gradient information, pass the ``retain_grad`` argument to
...
+.. doctest::
+
+   >>> y.backward()  # The default value of retain_grad is False
+   >>> z.grad is None
+   True
+
 All these computations are easily generalized to multi-element array input.
 Note that if we want to start backward computation from a variable holding a multi-element array, we must set the *initial error* manually.
-This is simply done by setting the :attr:`~Variable.grad` attribute of the output variable:
+Becuase when the :attr:`~Variable.size` of a variable (it means the number of elements in the array) is ``1``, it's considered as a variable object that represents a loss value, so that the :attr:`~Variable.grad` attribute of the variable is automatically filled with ``1``.
```
Comment (zori) [inline docs/source/tutorial/basic.rst]: Because ...

File: docs/source/tutorial/basic.rst [hunk 2]
```diff
@@ -41,6 +44,15 @@ We will show in this tutorial how to define networks dynamically.
 This strategy also makes it easy to write multi-GPU parallelization, since logic comes closer to network manipulation.
 We will review such amenities in later sections of this tutorial.
 
+Chainer represents a network as *an execution path on a computational graph*.
+A computational graph is a series of function applications, so that it can be described with multiple :class:`Function` objects.
+When such function is a layer of neural network, the parameters of the function will be updated through training.
+Therefore, the function needs to keep trainable parameters inside, so that Chainer has :class:`Link` class that can keep trainable parameters in the object of the class.
+The parameters of the function performed inside the :class:`Link` object are represented as :class:`Variable` objects.
+In short, the difference between these two objects, :class:`Link` and :class:`Function`, is basically whether it has trainable parameters inside or not.
```
Comment (zori) [inline docs/source/tutorial/basic.rst]: I propose `is whether it contains trainable parameters or not.`

File: docs/source/tutorial/basic.rst [hunk 3]
```diff
@@ -41,6 +44,15 @@ We will show in this tutorial how to define networks dynamically.
...
+A neural network model is typically described as a series of :class:`Function` and :class:`Link`.
+
+You can build a computational graph by dynamically 'chaining' various kinds of :class:`Link` and :class:`Function` to define a :class:`Chain`, and define a network by *running* the graph, so that it's called **Chainer**.
```
Comment (zori) [inline docs/source/tutorial/basic.rst]: I propose `to define a :class:`Chain`. In the framework the network is defined by *running* the chained graph, hence the name - **Chainer**.

File: docs/source/tutorial/basic.rst [hunk 4]
```diff
@@ -112,9 +124,19 @@ In order to preserve gradient information, pass the ``retain_grad`` argument to
...
+
+   >>> y.backward()  # The default value of retain_grad is False
+   >>> z.grad is None
+   True
+
+All these computations are easily generalized to multi-element array input.
 Note that if we want to start backward computation from a variable holding a multi-element array, we must set the *initial error* manually.
-This is done simply by setting the :attr:`~Variable.grad` attribute of the output variable:
+Because when the :attr:`~Variable.size` of a variable (it means the number of elements in the array) is ``1``, it's considered as a variable object that represents a loss value, so that the :attr:`~Variable.grad` attribute of the variable is automatically filled with ``1``.
+On the other hand, when the size of a variable is larger than ``1``, the :attr:`~Variable.grad` attribute remains ``None``, so that we need to give *initial error* explicitly by hand before :meth:`~Variable.backward`.
```
Comment (zori) [inline docs/source/tutorial/basic.rst]: remains ``None``, and it is necessary to set the *initial error* explicitly before running :meth:`~Variable.backward`.

**Comment by mitmul** [comment]
@zori Thank you for your review and sorry about this late reaction. I fixed the points you mentioned above, and ensured all Jenkints test jobs succeeded. Please merge it :)

**Comment by okuta** [comment]
LGTM!

**Pull request merged**